index.bs

```diff
@@ -91,66 +114,72 @@ dictionary CSSStyleSheetInit {
-		    * Otherwise, resolve |promise| with |sheet| once all of them have finished loading. and unset |sheet|'s [=disallow modification flag=].
-		8. Return |promise|.
+		4. Let |promise| be a promise.
+		5. [=In parallel=], do these steps:
+			1. Let |rules| be the result of running [=parse a list of rules=] from |text|.
+			2. Wait for loading of <a>@import</a> rules in |rules| and any nested <a>@import</a>s from those rules (and so on).
+				* If any of them failed to load, [=terminate=] fetching of the remaining <a>@import</a> rules,  and [=queue a task=] on the [=networking task source=] to perform the following steps:
+                    1. Unset |sheet|'s [=disallow modification flag=].
+                    2. Reject |promise| with reason set to "{{NotAllowedError}}" {{DOMException}}.
+			    * Otherwise, once  all of them have finished loading, [=queue a task=] on the [=networking task source=] to perform the following steps:
+                    1. Unset |sheet|'s [=disallow modification flag=].
+                    2. Set |sheet|'s [=CSS rules=] to |rules|.
+                    3. Resolve |promise| with |sheet|.
+
+				<p class="note">
+                    Note: Loading  of <a>@import</a> rules should follow the rules used for fetching style sheets for <a>@import</a> rules of stylesheets from &lt;link> elements, in regard to what counts as success, CSP, and Content-Type header checking.
+				</p>
+				<p class="note">
+					Note: We will use the [=fetch group=] of |sheet|'s [=constructor document=]'s [=relevant settings object=] for <a>@import</a> rules and other (fonts, etc) loads.
+				</p>
+				<p class="note">
+					Note: The rules regarding loading mentioned above are currently not specified rigorously anywhere.
+				</p>
+		6. Return |promise|.
 	</dd>
 
-    <dt><dfn method for=CSSStyleSheet lt="replaceSync(text)">replaceSync(text)</dfn></dt>
+
+    <dt><dfn method for=CSSStyleSheet>replaceSync(|text|)</dfn></dt>
 	<dd>
 		When called, execute these steps:
 
 		1. Let |sheet| be the stylesheet on which this function is called on.
 		2. If |sheet|'s [=constructed flag=] is not set, or |sheet|'s [=disallow modification flag=] is set, throw a "{{NotAllowedError}}" {{DOMException}}.
 		3. Set |sheet|'s [=CSS rules=] to an empty list, and set |sheet|'s [=disallow modification flag=].
-		4. <a spec=css-syntax-3>Parse a list of rules</a> from {{text}}.
+		4. [=Parse a list of rules=] from |text|.
 			If it returned a list of rules,
 			assign the list as |sheet|'s [=CSS rules=];
 			otherwise,
 			set |sheet|'s [=CSS rules=] to an empty list.
-		5. If |sheet| contains one or more <a spec=css-cascade-4>@import</a> rules, throw a "{{NotAllowedError}}" {{DOMException}}.
+		5. If |sheet| contains one or more <a>@import</a> rules, throw a "{{NotAllowedError}}" {{DOMException}}.
```

Comment:
If you throw this error, the flag never gets unset. Is that intentional?

If not, then the better design is probably not to touch the flag at all during this method, since it's sync anyway so no code could run that would modify it.